### PR TITLE
[FIX]  By default, when a task is linked to an activity that has a category defined, the task should inherit the category of the Activity

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -222,6 +222,8 @@ def onchange_parent_id(self):
                 self.responsible_id = self.parent_id.responsible_id
             if self.parent_id.partner_id:
                 self.partner_id = self.parent_id.partner_id
+            if self.parent_id.category_id:
```
Comment (eilst) [inline project_event/models/project_task.py:225]: Add tests for this new changes

**Review by eilst — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 1 above)
Comment (jananjoy) [reply project_event/models/project_task.py:225]: @eilst Done

File: project_event/tests/test_project_event_task.py [hunk 2]
```diff
@@ -168,3 +169,26 @@ def test_070_cancel_action(self):
         self.assertEqual(
             calendar_event.state,
             'cancelled')
+
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py:206]: Check all the previous test, there are some tests that did not pass

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 2 above)
Comment (eilst) [reply project_event/tests/test_project_event_task.py:206]: Yes, it fails some tests
2018-12-18 21:40:08,599 155 **ERROR** 11.0_dec_18_07 odoo.addons.project_event.tests.test_project_event: FAIL

File: project_event/tests/test_project_event_task.py [hunk 3]
```diff
@@ -168,3 +169,10 @@ def test_070_cancel_action(self):
...
+    def test_080_onchange_parent_id(self):
+        self.activity_1.onchange_parent_id()
+
```
Comment (eilst) [inline project_event/tests/test_project_event_task.py:243]: What about the children? The responsible, partner, and category of the children tasks?

Comment (eilst) [reply project_event/tests/test_project_event_task.py:243]: "Principal task should inherit the category of the Activity"

Comment (eilst) [reply project_event/tests/test_project_event_task.py:243]: Or the title is mistaken? right?

**Review by eilst — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 3 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:243]: @eilst Change the title based on task

Comment (eilst) [reply project_event/tests/test_project_event_task.py:243]: So you need to add a validation for: "the task should inherit the category of the Activity"

File: project_event/tests/test_project_event_task.py (hunk 2 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:206]: @rimbendhaou @eilst  Changed and re-tested it is running fine

File: project_event/tests/test_project_event_task.py (hunk 3 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:243]: @eilst don't understand this comment. Can you provide an example in the project code or existing test cases?

Comment (eilst) [reply project_event/tests/test_project_event_task.py:243]: You are trying to get the category of a list of children, you need to take one.

**Review by eilst — APPROVED** [review]

File: project_event/tests/test_project_event_task.py [hunk 4]
```diff
@@ -20,10 +20,11 @@ def setUp(self):
             'project_id': self.project_1.id,
             'responsible_id': self.project_1.responsible_id.id,
             'partner_id': self.project_1.partner_id.id,
+            'category_id': self.category_1.id,
             'room_id': self.room_1.id,
             'date_start': fields.Datetime.to_string(datetime.today()),
             'date_end': fields.Datetime.to_string(datetime.today() + timedelta(hours=4)),
-        })
+         })
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: @jananjoy indentation (extra space)

File: project_event/tests/test_project_event_task.py [hunk 5]
```diff
@@ -168,3 +169,17 @@ def test_070_cancel_action(self):
         self.assertEqual(
             calendar_event.state,
             'cancelled')
+
+    def test_080_onchange_parent_id(self):
+        self.activity_1.onchange_parent_id()
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: this is not the test of the onchange method, the test pass even if i comment the above line

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 5 above)
Comment (rimbendhaou) [reply project_event/tests/test_project_event_task.py]: This test must check the onchange_parent_id of a task not of an activity

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 4 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : changed it

File: project_event/tests/test_project_event_task.py (hunk 5 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: Done

File: project_event/tests/test_project_event_task.py [hunk 6]
```diff
@@ -20,6 +20,30 @@ def setUp(self):
...
             'partner_id': self.project_1.partner_id.id,
+            'category_id': self.category_1.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() + timedelta(hours=4)),
+         })
+        self.activity_2 = self.Tasks.create({
+            'name': 'Test Activity 1',
+            'activity_task_type': 'activity',
+            'project_id': self.project_1.id,
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: better to test on an orphan activity (no project_id)

File: project_event/tests/test_project_event_task.py [hunk 7]
```diff
@@ -168,3 +192,15 @@ def test_070_cancel_action(self):
         self.assertEqual(
             calendar_event.state,
             'cancelled')
+
+    def test_080_onchange_parent_id(self):
+        self.task_1.parent_id = self.activity_2
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py:268]: add the test of responsible and the partner

**Review by rimbendhaou — CHANGES_REQUESTED** [review]
rebase to have the latest modification

File: project_event/tests/test_project_event_task.py (hunk 7 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:268]: @rimbendhaou :Done

File: project_event/tests/test_project_event_task.py (hunk 6 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : Deleted the field name "project_id" from the self.activity_2 .

File: project_event/tests/test_project_event_task.py [hunk 8]
```diff
@@ -215,18 +236,42 @@ def test_090_create_orphan_task(self):
...
-        self.assertEqual(self.activity_1.name,'Test Activity 1')
+        self.assertEqual(self.activity_1.name, 'Test Activity 1')
 
     def test_110_compute_complete_name_task(self):
         complete_name = '%s / %s' % (self.task_2.code, self.task_2.name)
-        self.assertEqual(self.task_2.complete_name,complete_name)
+        self.assertEqual(self.task_2.complete_name, complete_name)
+
+    def test_120_onchange_parent_id(self):
+        self.task_1.parent_id = self.activity_2
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py:268]: activity_1 and activity_2 have the same responsible you must change the responsible of the activity_2 for example to another responsible (responsible_2 : you need to add it if it doesn't exist) and check if the responsible of the task is changed to the responsible2 when we change the parent from activity_1 to activity_2. (same thing for the partner)

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 7 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : What is your expected name ?

File: project_event/tests/test_project_event_task.py (hunk 8 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:268]: @rimbendhaou :Ok Changed as per your comments but the implemented functionality and corresponding test cases are completely different.

File: project_event/tests/test_project_event_task.py (hunk 6 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou Done

File: project_event/tests/test_project_event_task.py (hunk 3 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py:243]: @eilst  Done

File: project_event/tests/test_project_event_task.py [hunk 9]
```diff
@@ -20,11 +20,33 @@ def setUp(self):
...
             'partner_id': self.project_1.partner_id.id,
+            'category_id': self.category_1.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() + timedelta(hours=4)),
+        })
+        self.activity_2 = self.Tasks.create({
+            'name': 'Test Activity 2',
+            'activity_task_type': 'activity',
+            'responsible_id': self.project_2.responsible_id.id,
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: you can directly put 'responsible_id': self.responsible_2.id,

File: project_event/tests/test_project_event_task.py [hunk 10]
```diff
@@ -20,11 +20,33 @@ def setUp(self):
...
+            'partner_id': self.project_2.partner_id.id,
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: 'partner_id': self.partner_2.id,

File: project_event/tests/test_project_event_task.py [hunk 11]
```diff
@@ -215,18 +236,42 @@ def test_090_create_orphan_task(self):
...
+        self.assertEqual(self.task_1.category_id.name, 'Category 2')
+        self.task_1.parent_id = self.activity_1
+        self.assertEqual(self.task_1.responsible_id.name, 'Responsible 2')
+        self.assertEqual(self.task_1.partner_id.name, 'Partner 2')
+        self.assertEqual(self.task_1.category_id.name, 'Category 2')
+        self.task_1.onchange_parent_id()
+        self.assertEqual(self.task_1.responsible_id.name, 'Responsible 1')
+        self.assertEqual(self.task_1.partner_id.name, 'Partner 1')
+        self.assertEqual(self.task_1.category_id.name, 'Category 1')
+        self.assertEqual(self.activity_1.responsible_id.name, 'Responsible 1')
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: No need to do all these tests. onchange method will not affect the parent_id.responsible_id field

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 9 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: Done

File: project_event/tests/test_project_event_task.py (hunk 10 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @eilst Done

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**